src/passes/Print.cpp

```diff
@@ -242,6 +243,22 @@ struct PrintSExpression : public UnifiedExpressionVisitor<PrintSExpression> {
     return printPrefixedTypes("param", type);
   }
 
+  std::ostream& printBlockType(Signature sig) {
+    assert(sig.params == Type::none);
+    if (sig.results == Type::none) {
+      return o;
+    }
+    if (sig.results.isTuple()) {
+      if (auto it = signatureTypes.find(sig); it != signatureTypes.end()) {
+        o << "(type ";
+        printHeapType(it->second);
+        o << ") ";
```

Comment:
Do we need to print `(type $foo) (result i32 i32)`? Can't we just print `(result i32 i32)` without the type's name?

Reply:
Technically, if we only print `(result i32 i32)`, then that implicitly declares and uses this type if it does not otherwise exist in the module: `(rec (type (sub final (func (result i32 i32)))))).0`. By printing the real heap type we want to refer to, we avoid defining that new type.

This doesn't actually matter in practice because we don't store a heap type on control flow structures, just a signature, but it's good to emit text that more precisely matches what our binaries will look like. In particular, it would make a difference if you had some other tool parse our text format.